modules/core/src/lib/layer.ts

```diff
@@ -1072,10 +1080,23 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
 
       context.device.setParametersWebGL({polygonOffset: offsets});
 
+      const deckParameters = this.context.deck?.props.parameters;
+      const layerParameters = this.props.parameters;
+      const mergedParameters = {...deckParameters, ...layerParameters, ...parameters};
```

Comment:
`parameters` already contains `layer.props.parameters` (merged in LayersPass). I think this merging logic should reside in the pass entirely.